cmake/ProtoBuf.cmake

```diff
@@ -196,7 +196,7 @@ function(caffe2_protobuf_generate_cpp_py srcs_var hdrs_var python_var)
 
         # If we remove all reference to these pb.h files from external
         # libraries and binaries this rewrite can be removed.
-        COMMAND ${CMAKE_COMMAND} -DFILENAME=${CMAKE_CURRENT_BINARY_DIR}/${fil_we}.pb.h -DNAMESPACES=caffe\;caffe2\;onnx\;torch -P ${PROJECT_SOURCE_DIR}/cmake/ProtoBufPatch.cmake
+        COMMAND ${CMAKE_COMMAND} -DFILENAME=${CMAKE_CURRENT_BINARY_DIR}/${fil_we}.pb.h -DNAMESPACES=caffe\;caffe2\;onnx\;torch -DSYSTEM_PROTOBUF=OFF -P ${PROJECT_SOURCE_DIR}/cmake/ProtoBufPatch.cmake
```

Comment:
Suggest to unify these 2 lines to `-DSYSTEM_PROTOBUF=${CAFFE2_LINK_LOCAL_PROTOBUF}`

Reply:
Sounds reasonable.

Reply:
Wait, it should be !CAFFE2_LINK_LOCAL_PROTOBUF, right?

Reply:
Emmm yes.....then this make sense.

Reply:
OK the `LOCAL_PROTOBUF` approach is better.
BTW why not just pass in CAFFE2_LINK_LOCAL_PROTOBUF. grep experience is better that way